framework/src/main/java/org/checkerframework/framework/type/HashcodeAtmVisitor.java

```diff
@@ -31,8 +31,10 @@ protected Integer defaultAction(AnnotatedTypeMirror type, Void v) {
         if (type == null) {
             return 0;
         }
+        if (type.underlyingTypeHashcode == -1) {
+            type.underlyingTypeHashcode = type.underlyingType.toString().hashCode() * 13;
+        }
 
-        return type.getAnnotations().toString().hashCode() * 17
-                + type.getUnderlyingType().toString().hashCode() * 13;
+        return type.getAnnotations().toString().hashCode() * 17 + type.underlyingTypeHashcode;
```

Comment:
The previous code was in bad style because it attempted to reimplement hashing.  Please use `Objects.hash` instead.